kerchunk/codecs.py

```diff
@@ -106,7 +106,7 @@ def decode(self, buf, out=None):
                 if out is not None:
                     return numcodecs.compat.ndarray_copy(data, out)
                 else:
-                    return data.astype(dt)
+                    return data.astype(dt, copy=False)
```

Comment:
small 5% improvement. 81% of the time is now in `eccodes.codes_get_array`, I'm not sure theres any improvement to be had there.

Reply:
> 81% of the time is now in eccodes.codes_get_array, I'm not sure theres any improvement to be had there.

👍 

I don't think there's anything we can do without delving deep into the low-level codebase.

Reply:
`codes_get_array` does take a `ktype` so technically we could push the casting there. I'm not sure if it supports all the dtypes though.

[API doc](https://sites.ecmwf.int/docs/eccodes/namespaceec_codes.html#a648bae4beadb6a82698da40e08323be9)
```
ktype | the type we want the output in (can be int, float or string), native type if not specified
```

Reply:
Worth a try?

I have had a delve of the grib .cc code hoping to isolate the buffer find and decode and most admit to being lost in the many layers of obfuscation.

Reply:
Pushed a commit, there's no real difference on my test case because the dtypes are the same.